c/embeddedjs.c

```diff
@@ -1985,16 +1985,15 @@ JSModuleDef *ejsModuleLoader(JSContext *ctx,
     size_t buf_len;
     uint8_t *buf;
     JSValue func_val;
-    char *nameToLoad = (char*)moduleName;
+    char *nameToLoad = malloc(modNameLen + 4); // in case .js is not included
```

Comment:
could this be a `char nameToLoad[modNameLen+4]`? I just don't see a free, so this will add up.
Also, `safeMalloc()` rather than `malloc()` is preferred.

Reply:
> In general, I am not clear on what is being fixed here. Was the allocation length wrong, and more data got copied than should have been? And is it still wrong but moved to a heap allocation with safeMalloc/safeFree so it didn't trash the stack.
> 
> I would feel better if there were a case to reproduce this.

> In general, I am not clear on what is being fixed here. Was the allocation length wrong, and more data got copied than should have been? And is it still wrong but moved to a heap allocation with safeMalloc/safeFree so it didn't trash the stack.
> 
> I would feel better if there were a case to reproduce this.

I'm no C expert, but I think the issue is the declaration of the array within the `else` block, you can see in this screenshot the result of some logging I added:

```c
      char asciiExtendedName[modNameLen+3+1];
      memcpy(asciiExtendedName,moduleName,modNameLen);
      fprintf(f, "asciiExtendedName1: '%s'", asciiExtendedName);
      memcpy(asciiExtendedName+modNameLen,asciiDotJS,4);
      fprintf(f, "asciiExtendedName2: '%s'", asciiExtendedName);
      nameToLoad = asciiExtendedName;
    }
    fprintf(f, "nameToLoad: '%s'", nameToLoad);
    buf = js_load_file(ctx, &buf_len, nameToLoad);
```

<img width="671" alt="image" src="https://github.com/zowe/zowe-common-c/assets/3085193/557eba58-ad91-4c93-88c1-db87c5a1eaa0">
[embeddedjs.log](https://github.com/zowe/zowe-common-c/files/11675249/embeddedjs.log)

This issue is fixed in our installation even with moving the array declaration outside the if block (`      char asciiExtendedName[modNameLen+3+1];`) if this is preferable I can definitely do that.

Reply:
`7D` is the apostrophe in ebcdic, so the last use of the `nameToLoad` is wrong in the hex

Reply:
yes, done, thanks